### ISSUE TITLE
[MCU] UDS messageInterpreter (Rx)

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#43]
## User story
1. As a developer
2. I need a the UDS module to check what kind of message it is
3. So that the received messages are passed along to the correct module

## Acceptance criteria
* Messages are received from Communication Layer
* Mock Classes for Session Manager, Memory and Flashing are created
* Getting System Name (0xF197) of Read Data By Identifier (SID 0x22) is answered correctly
* Structure is ready to implement other kinds of messages

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
